library/client_events.js

```diff
@@ -117,17 +117,20 @@ module.exports = client => {
             client.log('Created new guild.')
           } else {
             // if in any practice channel, member has rights to speak in practice room chat (can be muted)
-            let prChan = client.channels.find(chan => chan.name === 'practice-room-chat')
+            let prChan = client.guilds.get(newMember.guild.id).channels.find(chan => chan.name === 'practice-room-chat')
             if (prChan == null) {
               client.log('Cannot find #practice-room-chat!')
             } else if (restwo.permitted_channels.includes(newMember.voiceChannelID)) {
               prChan.overwritePermissions(newMember.user, { SEND_MESSAGES: true })
             } else {
-              // remove the permission overwrite when member is no longer in a practice room
               let existingOverride = prChan.permissionOverwrites.get(newMember.user.id)
-              // this shouldn't happen unless someone manually deletes the override, but if for some reason it's gone, no big deal, just move on.
+              // existingOverride shouldn't be null unless someone manually deletes the override, but if for some reason it's gone, no big deal, just move on.
               if (existingOverride != null) {
-                existingOverride.delete()
+                if (existingOverride.allowed.bitfield === 0x800 && existingOverride.denied.bitfield === 0) { // the only permission was allow SEND_MESSAGES
```

Comment:
https://discord.js.org/#/docs/main/stable/class/Permissions?scrollTo=has suggests we could write:
`existingOverride.allowed.has("SEND_MESSAGES")` perhaps?

And also maybe, `existingOverride.denied.toArray().length === 0` or some such?

I would actually ignore the denied field since we don't modify that. We grant `SEND_MESSAGES` so could we just ... `existingOverride.allowed.remove('SEND_MESSAGES')`

Does that seem right?

Reply:
The issue with using ```existingOverride.allowed.has('SEND_MESSAGES')``` is that ```.has()``` only checks if it has those permissions, not the ```SEND_MESSAGES``` permission is the only permission allowed. Comparing the bitfield gives an exact match.

Basically we want it to set the ```SEND_MESSAGES``` permission to null in the overwrite. If that would cause every single permission to be null (i.e. nothing in both the deny and the allow lists), then it should delete the overwrite entirely, since it's functionally useless at that point. Short of doing a full array comparison I'm not sure how else to distinguish between "```SEND_MESSAGES``` was the only permission either allowed or denied so discard the overwrite entirely" and "there are other permissions in either the allow or deny list of the overwrite so just remove the ```SEND_MESSAGES``` allowance".

Incidentally, ```existingOverride.allowed.remove('SEND_MESSAGES')``` doesn't seem to have any effect - I suspect because the ```Permissions``` object is probably immutable.

Reply:
Looking at the source for ```PermissionsOverwrite.allowed``` reveals that it is indeed immutable, because the return value has ```freeze()``` called first.